$OpenBSD: patch-fswebcam_c,v 1.1 2020/10/19 16:50:16 sthen Exp $

From d5c708782bb947c9a610e413b4c7383dc10c388c Mon Sep 17 00:00:00 2001
From: Ashley Dixon <ash@suugaku.co.uk>
Date: Tue, 21 Jul 2020 02:33:50 +0100
Subject: [PATCH] Fixed usage of gdImageStringFT.

Since libgd 2.3.0, gdImageStringFT does not return NULL for an empty
string, thus convincing fswebcam that the font cannot be found and the
banner must be disabled.

https://github.com/libgd/libgd/issues/615

Index: fswebcam.c
--- fswebcam.c.orig
+++ fswebcam.c
@@ -467,7 +467,7 @@ int fswc_output(fswebcam_config_t *config, char *name,
 		char *err;
 		
 		/* Check if drawing text works */
-		err = gdImageStringFT(NULL, NULL, 0, config->font, config->fontsize, 0.0, 0, 0, "");
+		err = gdImageStringFT(NULL, NULL, 0, config->font, config->fontsize, 0.0, 0, 0, "A");
 		
 		if(!err) fswc_draw_banner(config, im);
 		else
